cmd/zpool/zpool_main.c

```diff
@@ -676,6 +676,16 @@ zpool_do_add(int argc, char **argv)
 		return (1);
 	}
 
+	if (!nvlist_exists(props, ZPOOL_CONFIG_ASHIFT)) {
+		char ashift[ZPOOL_MAXPROPLEN];
+		if (zpool_get_prop(zhp, ZPOOL_PROP_ASHIFT, ashift,
```

Comment:
You can use `zpool_get_prop_int()` since the property registered as a number with `zprop_register_number()`.